### PR TITLE
Feature 60 geopackage support

### PR DESCRIPTION
**Pull request by platonovsimeon**

### CODE REVIEW

File: app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import com.mapbox.mapboxsdk.style.expressions.Expression.get
+import com.mapbox.mapboxsdk.style.layers.FillExtrusionLayer
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory.*
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource
+import fi.metatavu.muisti.api.client.models.PageLayoutViewProperty
+import fi.metatavu.muisti.exhibitionui.BuildConfig
+import fi.metatavu.muisti.exhibitionui.pages.PageViewLifecycleListener
+import fi.metatavu.muisti.exhibitionui.views.PageActivity
+
+class MapComponentContainer(buildContext: ComponentBuildContext): FrameLayout(buildContext.context) {
```
Comment (anttileppa) [inline app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt:23]: Add docs

File: app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    init {
```
Comment (anttileppa) [inline app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt:25]: Add docs

File: app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                else -> super.setProperty(buildContext, parent, view, property)
+            }
+        } catch (e: Exception) {
+            Log.d(PlayerViewComponentFactory::javaClass.name, "Failed to set property ${property.name} to ${property.value}}", e)
+        }
+    }
+
+}
+
+private class MapViewLifeCycleListener(val buildContext: ComponentBuildContext, val view: MapComponentContainer): PageViewLifecycleListener, OnMapReadyCallback {
```
Comment (anttileppa) [inline app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt:68]: Add docs

File: app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+class MapComponentContainer(buildContext: ComponentBuildContext): FrameLayout(buildContext.context) {
+
+    init {
+        buildContext.addLifecycleListener(
+            MapViewLifeCycleListener(buildContext,this)
+        )
+    }
+
+
```
Comment (anttileppa) [inline app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt:31]:
```suggestion
```

File: app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    private lateinit var mapView: MapView
+    private lateinit var geoJsonSource: GeoJsonSource
+    private lateinit var mapboxMap: MapboxMap
+
+    override fun onPageActivate(pageActivity: PageActivity) {
+        val context: Context = pageActivity
+        Mapbox.getInstance(context, BuildConfig.MAP_BOX_ACCESS_TOKEN)
+
+        val mapView = MapView(context)
```
Comment (anttileppa) [inline app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt:78]: Does MapView need handlers for e.g. out of memoty?

File: app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    override fun onMapReady(mapboxMap: MapboxMap) {
+        this.mapboxMap = mapboxMap
+        this.mapboxMap.setStyle(Style.OUTDOORS) {
+            @Override
+            fun onStyleLoaded(@NonNull style: Style) {
+                style.addSource(geoJsonSource)
+                style.addLayer(
+                    FillExtrusionLayer("deaths", "deaths").withProperties(
```
Comment (anttileppa) [inline app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt:96]: deaths?

Comment (platonovsimeon) [reply app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt:106]: This is based on the previously used GeoJsonSource

File: app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource
+import fi.metatavu.muisti.api.client.models.PageLayoutViewProperty
+import fi.metatavu.muisti.exhibitionui.BuildConfig
+import fi.metatavu.muisti.exhibitionui.pages.PageViewLifecycleListener
+import fi.metatavu.muisti.exhibitionui.views.PageActivity
+
+/**
+ * Component container for initializing life cycle-listener for Map View
+ */
+class MapComponentContainer(buildContext: ComponentBuildContext): FrameLayout(buildContext.context) {
```
Comment (MrWalrus3451) [inline app/src/main/java/fi/metatavu/muisti/exhibitionui/pages/components/MapViewComponentFactory.kt:27]:
```suggestion
/**
 * Component container for initializing life cycle-listener for Map View
 * @param buildContext Component Build Context
 */
class MapComponentContainer(buildContext: ComponentBuildContext): FrameLayout(buildContext.context) {
```

**Pull request merged**